File: BinarySearch.cpp
#include <iostream>

using namespace std;

int BinarySearch(int a[],int x,int n)
{
	int left = 0;
	int right = n - 1;
	while (left <= right)
	{
		int middle = (left + right)/2;
		if (x == a[middle])
			return middle;
		if (x > a[middle])
			left = middle + 1;
		else
			right = middle - 1;
	}
	return -1;
 } 
 
int main(){
	int nums[7] = {5,12,23,45,66,78,98};
	cout << BinarySearch(nums,66,7) <<endl;
	return 0;
}

File: 17计科一班 潘咏聪 170201104724/BinarySearch.java
import java.util.Scanner;

public class BinarySearch {
    public static void main(String[] args){
        int a[]={1,2,3,4,5,6,7,8,9};
        int left=0;
        int right=8;
        int x;
        int flag=0;
        System.out.println("输入你要查找的数字：");
        Scanner scan = new Scanner(System.in);
        x=scan.nextInt();
        while (left<=right){
            int middle=(left+right)/2;
            if(x==a[middle]){
                flag=1;
                System.out.println("在数组中的位置为：");
                System.out.println(middle);
                break;
            }
            if(x>a[middle]){
                left=middle+1;
            }
            else {
                right=middle-1;
            }
        }
        if(flag==0){
            System.out.println("没找到该数字");
        }
    }
}
